napari/util/tests/test_keybindings.py

```diff
@@ -95,61 +104,183 @@ def foo():
     assert kb == dict(A=foo)
 
 
-def test_InheritedKeymap():
-    d = {}
-    k = InheritedKeymap(lambda: d, {'A': 0})
-    assert k['A'] == 0
+def test_keymap_provider():
+    class Foo(KeymapProvider):
+        ...
 
-    d['B'] = 1
-    assert 'B' in k
-    assert k['B'] == 1
+    assert Foo.class_keymap == {}
 
-    assert k.copy()['A'] == 0
-    assert k.copy()['B'] == 1
+    foo = Foo()
+    assert foo.keymap == {}
+
+    class Bar(Foo):
+        ...
+
+    assert Bar.class_keymap == {}
+    assert Bar.class_keymap is not Foo.class_keymap
 
-    k['B'] = 2
-    assert k['B'] == 2
+    class Baz(KeymapProvider):
+        class_keymap = {'A', ...}
 
-    assert k.pop('B') == 2
-    assert k.data['B'] is None
-    with pytest.raises(KeyError):
-        k['B']
+    assert Baz.class_keymap == {'A', ...}
 
 
-def test_KeymapMixin():
-    class Foo(KeymapMixin):
+def test_keymap_handler():
+    class Foo(KeymapProvider):
+        class_keymap = {
+            'A': lambda x: setattr(x, 'A', ...),
+            'B': lambda x: setattr(x, 'B', ...),
+            'C': lambda x: setattr(x, 'C', ...),
+            'D': ...,
+        }
+
+    foo = Foo()
+    foo.keymap = {
+        'B': lambda x: setattr(x, 'B', None),  # overwrite
+        'E': lambda x: setattr(x, 'E', None),  # new entry
+        'C': ...,  # blocker
+    }
+
+    handler = KeymapHandler()
+    handler.keymap_providers.insert(0, foo)
+
+    assert handler.keymap_chain.maps == [foo.keymap, foo.class_keymap]
+    assert handler.active_keymap == {
+        'A': foo.class_keymap['A'],
+        'B': foo.keymap['B'],
+        'E': foo.keymap['E'],
+    }
+
+    # non-overwritten class keybinding
+    assert not hasattr(foo, 'A')
+    handler.press_key('A')
+    assert foo.A is ...
+
+    # keybinding blocker on class
+    handler.press_key('D')
+    assert not hasattr(foo, 'D')
+
+    # non-overwriting instance keybinding
+    assert not hasattr(foo, 'E')
+    handler.press_key('E')
+    assert foo.E is None
```

Comment:
For each of these, would you mind expanding the comment to explain the resolution path relative to the object and keymap creation above? e.g. "the keymap for E is defined in the Foo instance definition, which is the first keymap checked, before the class keymap."